### PR TITLE
Customizable Payment Options: Implement a user-friendly interface on the BLT website for users to select their preferred cryptocurrency, input the amount they wish to pay or donate, and enter a unique payment address or QR code for the transaction. #1912

### PR DESCRIPTION
**Pull request by Sarthak5598**
#1912

### CODE REVIEW

**Comment by Sarthak5598** [comment]
https://www.loom.com/share/852cf10e68494bedbc7aef413c8e84ef?sid=f1ed41b3-ee08-41b6-8794-328edfcd82c8
This is a link to video showing all the changes I made please check it out @DonnieBLT @AtmegaBuzz

**Comment by AtmegaBuzz** [comment]
ArcoLinux_2024-03-07_00-01-43
Can you make the update address UI something like this.

Also i didn't got how you are switching between ethereum <-> bitcoin qr code. 
if you can add some arrow above the qr code to switch on click that would be noticeable for the user and they will know that they can see other address too.

Rest looks awesome, ill review it once this is done, will also see what the problem with double click

**Comment by Sarthak5598** [comment]
Okay sure !

**Comment by Sarthak5598** [comment]
image
image
@AtmegaBuzz 
CHECK THIS OUT

**Comment by AtmegaBuzz** [comment]
Don't you think that arrow is hiding QR data, I suggested that you should move that above the qr code, something like this.
ArcoLinux_2024-03-07_15-15-59
 qr code

**Comment by Sarthak5598** [comment]
Screenshot 2024-03-07 220416
Screenshot 2024-03-07 220435
Check these out @AtmegaBuzz @DonnieBLT

**Comment by Sarthak5598** [comment]
Okay will work on it!!!

File: website/views.py [hunk 1]
```diff
@@ -3993,17 +3993,23 @@ def trademark_detailview(request, slug):
...
+        if selected_crypto and new_address:
             try:
                 user_profile = request.user.userprofile
-                user_profile.crypto_address = new_address
+                if selected_crypto == "Bitcoin":
+                    user_profile.btc_address = new_address
+                elif selected_crypto == "Ethereum":
+                    user_profile.eth_address = new_address
+                elif selected_crypto == "BitcoinCash":
+                    user_profile.bch_address = new_address
```
Comment (arkid15r) [inline website/views.py:4006]: Optional: it looks like a candidate for `match` statement (https://peps.python.org/pep-0636/#appendix-a-quick-intro)

Comment (Sarthak5598) [reply website/views.py:4006]: Okay, is match statement better than if else ?

Comment (arkid15r) [reply website/views.py:4006]: It may be more readable, I'm leaving it to you to decide whether it's better or not.

Comment (Sarthak5598) [reply website/views.py:4006]: image
I made the changes , It is better if else statement had repetition

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**